### PR TITLE
Use native video call picture in picture!

### PR DESCRIPTION
**Pull request by pixlwave**
This PR uses an `AVPictureInPictureVideoCallController` to PiP the whole EC web view instead of the previous solution where EC would request the web view to PiP the speaker's video. I've made the feature flag user configurable now as it no longer requires a specific EC deployment*.

This PR doesn't handle responses from the web view about whether PiP is possible so it is possible to get stuck in a call when an error occurs.

The second and third commits improves the handling of whether or not the we should start the PiP or dismiss the screen.

The fourth commit hides the call button when you're already joined to the call in that room.

\* It subsequently does now, however when using an unsupported deployment it will simply navigate back which isn't the case with the old implementation on `develop`. The user can enter https://pr2573--element-call.netlify.app/ as the base URL above the flag for testing purposes.

### CODE REVIEW

File: ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift [hunk 1]
```diff
@@ -580,7 +580,7 @@ class UserSessionFlowCoordinator: FlowCoordinatorProtocol {
     
     private var callScreenPictureInPictureController: AVPictureInPictureController?
     private func presentCallScreen(configuration: ElementCallConfiguration) {
-        guard elementCallService.ongoingCallRoomID != configuration.callID else {
+        guard elementCallService.ongoingCallRoomID != configuration.callRoomID else {
```
Comment (pixlwave) [inline ElementX/Sources/FlowCoordinators/UserSessionFlowCoordinator.swift:583]: unrelated refactor on a previous PR, this just seems a bit more correct given they're strings.

File: ElementX/Sources/Screens/CallScreen/View/CallScreen.swift [hunk 2]
```diff
@@ -46,52 +47,60 @@ struct CallScreen: View {
...
                 .id(context.viewState.url)
                 .ignoresSafeArea(edges: .bottom)
         }
     }
 }
 
-private struct WebView: UIViewRepresentable {
+private struct CallView: UIViewRepresentable {
+    /// The top-level view this representable displays. It wraps the web view when picture in picture isn't running.
+    class WebViewWrapper: UIView { }
```
Comment (pixlwave) [inline ElementX/Sources/Screens/CallScreen/View/CallScreen.swift:60]: Honestly not sure on this. I think it's nicer to have a subclass to help explain what the `UIViewRepsentable` is generic over (and that it's not the actual web view). But equally it's redundant seeing as this view doesn't need to know about the web view.

Comment (Velin92) [reply ElementX/Sources/Screens/CallScreen/View/CallScreen.swift:60]: Maybe you could just use a typealias

File: ElementX/Sources/Screens/CallScreen/View/CallScreen.swift [hunk 3]
```diff
@@ -120,6 +129,21 @@ private struct WebView: UIViewRepresentable {
             webView.isOpaque = false
             webView.backgroundColor = .compound.bgCanvasDefault
             webView.scrollView.backgroundColor = .compound.bgCanvasDefault
+            
+            webViewWrapper.addMatchedSubview(webView)
```
Comment (Velin92) [inline ElementX/Sources/Screens/CallScreen/View/CallScreen.swift:133]: actually why do we need the wrapper? what would prevent the WebView to be instantiated before `super.init()` and just have its delegates set to self after `super.init()`

**Review by Velin92 — APPROVED** [review]
LGTM, just a few suggestions/nits

File: ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift [hunk 4]
```diff
@@ -175,8 +175,10 @@ struct RoomScreen: View {
         
         if !ProcessInfo.processInfo.isiOSAppOnMac {
             ToolbarItem(placement: .primaryAction) {
-                callButton
-                    .disabled(roomContext.viewState.canJoinCall == false)
+                if roomContext.viewState.shouldShowCallButton {
+                    callButton
+                        .disabled(roomContext.viewState.canJoinCall == false)
```
[the remark this answers is not in the document]
Comment (pixlwave) [reply ElementX/Sources/Screens/RoomScreen/View/RoomScreen.swift:180]: Oh yeah good spot, I wonder why this isn't caught by SwiftLint/Format 🤔

File: ElementX/Sources/Screens/CallScreen/View/CallScreen.swift (hunk 3 above)
Comment (pixlwave) [reply ElementX/Sources/Screens/CallScreen/View/CallScreen.swift:133]: It's because a view can only be added to a single parent, so when the PiP starts and we add it to the PiPVC, it has to be removed from the SwiftUI view hierarchy. So it can't be the base view for the UIViewRepresentable as that doesn't have a way to let you take the view away from it.

Will try the type alias.
